lib/_http_server.js

```diff
@@ -217,6 +217,11 @@ function writeHead(statusCode, reason, obj) {
     // This class of status code indicates a provisional response,
     // consisting only of the Status-Line and optional headers, and is
     // terminated by an empty line.
+
+    if(statusCode == 304 && (headers.hasOwnProperty('trailer') || headers.hasOwnProperty('Trailer')){
```

Comment:
@jasnell Not true if the user is passing a headers object and no previous headers were set.

I think this kind of check is better placed in _http_outgoing.js in `_storeHeader()` where we already check for the Trailer header. Doing the relevant check there would allow us to avoid any lookups, we would just have `statusCode === 304 && state.trailer` as the conditional.